### PR TITLE
[Do Not Merge] Triton mlir ext dep fix

### PR DESCRIPTION
**Pull request by dllehr-amd**

### CODE REVIEW

File: python/triton/compiler.py [hunk 1]
```diff
@@ -1492,12 +1492,16 @@ def rocm_path_dir():
 def _build(name, src, srcdir):
     if torch.version.hip is not None:
         hip_lib_dir = os.path.join(rocm_path_dir(), "lib")
-        hip_include_dir = os.path.join(rocm_path_dir(), "include")
+        #hip_include_dir = os.path.join(rocm_path_dir(), "include")
+        base_dir = os.path.dirname(__file__)
+        rocm_path = os.path.join(base_dir, "third_party", "rocm")
+        hip_include_dir = os.path.join(rocm_path, "include")
+        #hip/hip_runtime.h
```
Comment (micmelesse) [inline python/triton/compiler.py:1499]: ditto

File: lib/Target/HSACO/HSACOTranslation.cpp [hunk 2]
```diff
@@ -133,7 +134,15 @@ std::string generate_hsaco(llvm::Module *module, const std::string &triple,
   std::string hsaco_path = kernel_name + std::string(".hsaco");
 
   std::string error_message;
-  std::string lld_path = ::triton::tools::getenv("ROCM_PATH") + "/llvm/bin/ld.lld";
+  // std::string lld_path = ::triton::tools::getenv("ROCM_PATH") + "/llvm/bin/ld.lld";
```
Comment (micmelesse) [inline lib/Target/HSACO/HSACOTranslation.cpp:137]: Clean up commented out code

**Review by micmelesse — APPROVED** [review]
Looks good. Just clean up comments.

**Comment by dllehr-amd** [comment]
I'm going to leave this PR open, as we need to solve the `libhsa-runtime64.so` problem before we are fully merged.  Until then we need to rely on ROCm being installed prior to executing.  Since that's a requirement, I also want to sort out a potentially better way of bringing with the header files, as we are currently at 16 of them, for only needing the hipModuleLaunchKernel

**Comment by dllehr-amd** [comment]
Closing as we're working to merge in #190 and #195

**Pull request closed without merging**